_posts/2025-04-09-phased-strimzi-upgrade-example.md

```diff
@@ -0,0 +1,733 @@
+
+As with any Kubernetes resource, Strimzi CRDs have an API version, which is currently [`v1beta2`](https://github.com/strimzi/strimzi-kafka-operator/blob/ac5f9b48db851737aba12abebee4000295d2ec1c/install/cluster-operator/040-Crd-kafka.yaml#L23).
+Theoretically, Strimzi managed CRs that share this API version can coexist and you _could_ run multiple versions concurrently on the same Kubernetes cluster (0.43. 0.44 ad 0.45 for example).
+However, the only guaranteed and tested configuration is two adjacent Strimzi versions.
+This has to be supported to allow the upgrade of a single Strimzi instance and so is tested as part of a Strimzi release. 
+If you install more than two adjacent Strimzi versions, you are running the risk of newer CRDs causing issues for older CRs. 
+For example, CRs managed by Strimzi 0.43 may fail if 0.45 CRDs are installed, as this is an unsupported and untested configuration.
+Therefore, it is strongly recommended that you only run two adjacent Strimzi versions using a system like the one proposed in this post.
+
+###### CRD API version upgrades
+
+The next consideration is when the CRD API version is upgraded.
+For example when Strimzi eventually upgrades from `v1beta2` to `v1` CRDs.
+When this occurs you will need to first make sure **all** Strimzi managed CRs are updated to the latest Strimzi version running the old CRD API version (e.g. `v1beta1`).
+You will need to do this **before** you install the CRDs for the new Strimzi version which uses the new CRD API version (e.g. `v1`).
+You would then follow any specific upgrade instructions and move each CR onto the new Strimzi version and CRD API version.
+
+Using the last CRD API version upgrade as an example, we can illustrate the process. 
+Strimzi 0.21 supports the older `v1beta1` API version, Strimzi 0.22 supports both the older `v1beta1` as well as the newer `v1beta2` and finally, Strimzi 0.23 supports only the new `v1beta2` CRDs. 
+Because of this, we can run Strimzi 0.21 and 0.22 concurrently as both support the older `v1beta1` API. 
+We can also run Strimzi 0.22 and 0.23 concurrently, as both support the newer `v1beta2` API. 
+However, we cannot run Strimzi 0.21 and 0.23 concurrently, as each supports a different API version.
+
+So, if we were running a setup with Strimzi 0.21 and wanted to upgrade to 0.23, we have to:
+
+1. First install CRDs from Strimzi 0.22 with both `v1beta1` and `v1beta2` API versions and install the Stirmzi 0.22 deployment, RBAC resources, etc.
+1. Migrate all Strimzi managed CRs from 0.21 to 0.22.
+1. Remove the 0.21 operator.
+1. Perform the specific [CRD upgrade instructions](https://strimzi.io/docs/operators/0.23.0/deploying#assembly-upgrade-resources-str), install CRDs from Strimzi 0.23 with only the `v1beta2` API versions and install the Stirmzi 0.23 deployment, RBAC resources, etc.
+1. Migrate all Strimzi managed CRs from 0.22 to 0.23.
+1. Remove the 0.22 operator.
+
+##### Strimzi install resources
```

Comment:
`Strimzi install resources` sounds strange to me. Feel free to keep it if it is intentional and makes sense to you. But maybe `Strimzi installation resources` would be better?

Reply:
Yes it does sound wrong, particularly because everywhere else in the doc apart from this title I have called it "Strimzi installation resources"?